AD74412R/AD74413R driver

CiprianRegus opened the pull request:
Implemented a driver and demo project for the AD74413R and AD74412R ADC-DAC devices. 

Draft until better tested on STM32.

amiclaus reviewed (commented):
some comments prior to `ready for review` stage.

amiclaus commented on drivers/adc-dac/ad74413r/ad74413r.h:
```diff
@@ -0,0 +1,502 @@
+#define AD74413R_ADC_MAX_VALUE		65535
+
+/** Register map */
+
```
nit: extra line

amiclaus commented on drivers/adc-dac/ad74413r/ad74413r.h:
```diff
@@ -0,0 +1,502 @@
+#define AD74413R_DEBOUNCE_MODE_MASK             NO_OS_BIT(5)
+#define AD74413R_DIAG_RESULT_MASK		NO_OS_GENMASK(15, 0)
+#define AD74413R_REV_ID				NO_OS_GENMASK(7, 0)
+
```
nit: extra line

amiclaus commented on projects/ad74413r/src/platform/maxim/main.c:
```diff
@@ -0,0 +1,73 @@
+
+	return ret;
+}
+
```
nit: extra line

amiclaus commented on drivers/adc-dac/ad74413r/ad74413r.c:
```diff
@@ -0,0 +1,1050 @@
+	switch (rejection) {
+	case AD74413R_REJECTION_50_60:
+		*rate = AD74413R_ADC_SAMPLE_20HZ;
+		return 0;
```
ditto.

CiprianRegus commented on drivers/adc-dac/ad74413r/ad74413r.c:
```diff
@@ -0,0 +1,1050 @@
+		delay = conv_times_ad74412r[rejection];
+
+	/** Wait for all channels to complete the conversion. */
+	if (delay < 1000)
```
It's the amount of microseconds in a millisecond. The elements of the `conv_times_ad74413r` array are microsecond delay values, which are present in the datasheet.

CiprianRegus commented:
- Implemented review change requests.
- Added the `ad74413r_reg_read_raw` function, which is used to read a raw comm frame. This will be useful for the IIO driver.

rbolboac commented on projects/ad74413r/builds.json:
```diff
@@ -0,0 +1,15 @@
+{
+  "stm32": {
+    "dummy_example": {
+      "flags" : "DUMMY_EXAMPLE=y IIO_EXAMPLE=n IIO_TRIGGER_EXAMPLE=n"
```
are IIO_EXAMPLE=n IIO_TRIGGER_EXAMPLE=n flags needed?

CiprianRegus replied:
They are not. Will add them back once the IIO example is implemented.

rbolboac commented on projects/ad74413r/builds.json:
```diff
@@ -0,0 +1,15 @@
+  },
+  "maxim": {
+    "dummy_example": {
+      "flags" : "DUMMY_EXAMPLE=y IIO_EXAMPLE=n IIO_TRIGGER_EXAMPLE=n TARGET=max78000"
```
are IIO_EXAMPLE=n IIO_TRIGGER_EXAMPLE=n flags needed?

rbolboac commented on projects/ad74413r/builds.json:
```diff
@@ -0,0 +1,15 @@
+      "flags" : "DUMMY_EXAMPLE=y IIO_EXAMPLE=n IIO_TRIGGER_EXAMPLE=n TARGET=max78000"
+    },
+    "dummy_example": {
+      "flags" : "DUMMY_EXAMPLE=y IIO_EXAMPLE=n IIO_TRIGGER_EXAMPLE=n TARGET=max32655"
```
are IIO_EXAMPLE=n IIO_TRIGGER_EXAMPLE=n flags needed?

rbolboac commented on projects/ad74413r/src.mk:
```diff
@@ -0,0 +1,39 @@
+		$(INCLUDE)/no_os_units.h
+
+SRCS += $(DRIVERS)/api/no_os_gpio.c \
+		$(DRIVERS)/api/no_os_i2c.c  \
```
is i2c needed?

rbolboac reviewed (commented):
looks good, some small questions on my side.

CiprianRegus commented:
- Removed unused CFLAGS.
